src/main/java/TaskManager.java

```diff
@@ -0,0 +1,91 @@
+import java.util.ArrayList;
+
+public class TaskManager {
+    protected ArrayList<Task> tasks;
+
+    public TaskManager() {
+        tasks = new ArrayList<>();
+    }
+
+    public void executeUserInput(String userInput) {
+        String command;
+        if (userInput.contains(" ")) {
+            command = userInput.substring(0, userInput.indexOf(" "));
+        } else {
+            command = userInput;
+        }
+
+        if (!validateInput(command, userInput)) {
+            Ui.printPretty("Sorry, couldn't recognize this input format");
+            return;
+        }
+
+        if (command.equals("list")) {
+            listTasks();
+        } else if (command.equals("todo") || command.equals("deadline") || command.equals("event")) {
+            Task newTask = getTaskFromUserInput(command, userInput);
+            if (newTask == null) {
+                Ui.printPretty("Sorry, couldn't recognize this input format");
+            } else {
+                addTask(newTask);
+            }
+        } else {
+            Ui.printPretty("Sorry, couldn't recognize this input format");
+        }
+    }
+
+    protected boolean validateInput(String command, String userInput) {
+        if (command.equals("todo")) {
+            return ToDo.validateUserInput(userInput);
+        } else if (command.equals("deadline")) {
+            return Deadline.validateUserInput(userInput);
+        } else if (command.equals("event")) {
+            return Event.validateUserInput(userInput);
+        } else if (command.equals("list")) {
+            return true; // If command is list then anything after doesn't affect functionality
```

Comment:
Comments should be indented relative to their position in the code
eg:
// If command is list then anything after doesn't affect functionality
return true;